lib/client.ts

```diff
@@ -35,23 +35,45 @@ function defaultFormatter(response: Response) {
   return Promise.resolve({});
 }
 
-const baseRequest = (defaults: RequestInit): Function => (url: string, config: RequestInit, reqConfig?: RequestConfig): Function => {
+const baseRequest = (defaults: RequestInit, options: ClientOptions): Function => (url: string, config: RequestInit, reqConfig?: RequestConfig): Function => {
   const headers = new Headers();
   headers.append('Accept', 'application/json');
   headers.append('Content-Type', 'application/json');
   headers.append('Procore-Sdk-Version', sdkVersionHeader);
   headers.append('Procore-Sdk-language', 'javascript');
 
-  let opts: RequestInit = { mode: 'cors', credentials: 'include', headers, ...defaults, ...config };
+  if (reqConfig?.companyId) {
+    headers.append('Procore-Company-Id', `${reqConfig.companyId}`);
+  } else if (options?.defaultCompanyId) {
+    headers.append('Procore-Company-Id', `${options.defaultCompanyId}`);
+  }
```

Comment:
@edwlan I thought some more about the warning. The customer could pass in a custom 'Procore-Company-Id' header instead of using the companyId or defaultCompanyId parameters. If we want to add a warning we would need to add it after line 61 and check if the Procore-Company-Id header is present.

Reply:
The change would look something like this:
![image](https://user-images.githubusercontent.com/109704931/189405191-f5529321-f1f7-4dd6-a008-f138d0bc2b11.png)

Reply:
One downside of this is that a couple APIs don't need a company_id, so this is technically incorrect.

Reply:
@edwlan Let's leave the warning out for now.